tests/decorators/test_clickjacking.py

```diff
@@ -0,0 +1,142 @@
+from asgiref.sync import iscoroutinefunction
+
+from django.http import HttpRequest, HttpResponse
+from django.middleware.clickjacking import XFrameOptionsMiddleware
+from django.test import SimpleTestCase
+from django.views.decorators.clickjacking import (
+    xframe_options_deny,
+    xframe_options_exempt,
+    xframe_options_sameorigin,
+)
+
+
+class XFrameOptionsDenyTests(SimpleTestCase):
+    """
+    Tests for the xframe_options_deny decorator.
+    """
+
+    def test_wrapped_sync_function_is_not_coroutine_function(self):
+        def sync_view(request):
+            return HttpResponse()
+
+        wrapped_view = xframe_options_deny(sync_view)
+        self.assertIs(iscoroutinefunction(wrapped_view), False)
+
+    def test_wrapped_async_function_is_coroutine_function(self):
+        async def async_view(request):
+            return HttpResponse()
+
+        wrapped_view = xframe_options_deny(async_view)
+        self.assertIs(iscoroutinefunction(wrapped_view), True)
+
+    def test_decorator_sets_x_frame_options_to_deny(self):
+        @xframe_options_deny
+        def a_view(request):
+            return HttpResponse()
+
+        response = a_view(HttpRequest())
+        self.assertEqual(response.headers["X-Frame-Options"], "DENY")
+
+    async def test_decorator_sets_x_frame_options_to_deny_with_async_view(self):
+        @xframe_options_deny
+        async def an_async_view(request):
+            return HttpResponse()
+
+        response = await an_async_view(HttpRequest())
+        self.assertEqual(response.headers["X-Frame-Options"], "DENY")
+
+
+class XFrameOptionsSameoriginTests(SimpleTestCase):
+    """
+    Tests for the xframe_options_sameorigin decorator.
+    """
```

Comment:
IMO such docstrings are unnecessary as a test class name is self-explanatory.

For the future, to add more explanation, docstrings should state the expected behavior and omit prefixes like "Tests for" since all tests test things. This guideline is from [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).